README.md

```diff
@@ -1,5 +1,73 @@
-# Insight_Project_Framework
-Framework for machine learning projects at Insight Data Science.
+# Vulnerable Building Detection
+A soft-story building is a multiple-story building in which the first floor is "soft", meaning it consists mostly of garage door openings or windows or other openings.  Soft-story buildings are particularly vulnerable to earthquakes, and thus the ability to quickly inventory soft-story buildings in an area may be of use to city or state officials.  This repo contains code to create and train a custom Inception V3 image classifier to detect soft-story buildings and deploy the model in a user-friendly Streamlit app.
+
+##Overview
+The Streamlit app takes as input from the user latitude and longitude coordinates specifying a bounding box region of the world and also a number of addresses.  It takes random latitude and longitude locations within that bounding box and uses reverse geocoding to obtain the addresses.  Then it uses those addresses to obtain Google Street View images.  The images are then sent through the image classifier and the results are presented to the user: "Soft-Story" means that the image is of a soft-story building, "Non-Soft-Story" means that the image is of a non-soft-story buildng, and "Bad Image" means that the image does not contain any buildings or it is unclear which building the image is supposed to refer to.
+
+There are two types of models that can be used:
+
+Ternary classification:  A single model classifies the images into the three categories.
+
+Binary classification:  One model classifies the image as good or bad, and then a second model classifies the image as soft or non-soft.
+
+##Installation and Dependencies
+Clone the repo
+
+##Preparing the Environment
+Before running any scripts in this repo, source the "build/environment.sh" bash script to add "Modules" to the PYTHONPATH environment variable:
+
+```
+source build/environment.sh
+```
```

Comment:
Ideally, you don't want to do this kind of "sys path hacking". It makes your code very fragile to small changes. For instance, in your script you assume that the user has a specific folder name (`$HOME/Insight/Vulnerable_Buildings`) - you shouldn't assume this. 

Also, this path hacking makes your code less modular - if another code would like to use that code and import it, it wouldn't work. 

Here is a better way: 
 * Put your main (highest level) executables (train, preprocess etc) in the top level of your directory and then use module imports in your file, i.e.
`from Modules import addresses` 

This is not a high priority - but keep it in mind in the future or if there is time left.